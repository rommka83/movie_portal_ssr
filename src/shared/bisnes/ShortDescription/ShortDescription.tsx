import React, { FC, HTMLAttributes, PropsWithChildren } from 'react';
import styles from './shortdescription.module.css';
import { IFilm } from 'shared/types/IFilm';
import classNames from 'classnames';
import { GenreBookmarks } from '../GenreBookmarks';

interface IProps {
  obj: IFilm;
  allGenres?: boolean;
}

export const ShortDescription: FC<PropsWithChildren<HTMLAttributes<HTMLDivElement> & IProps>> = ({
  obj,
  allGenres = false,
  className,
}) => {
  const countries = obj.countries.map((el) => el.name);

  const hours = Math.floor(obj.movieLength / 60);
  const min = obj.movieLength - hours * 60;
  return (
    <div className={classNames(styles.root, className)}>
      <p className={styles.yearСountry}>
        {obj.year}, {countries.join(', ')}
      </p>
      <div className={styles.genre}>
        {allGenres ? (
          <GenreBookmarks className={styles.genreBookmarks} ganre={obj.genres} />
        ) : (
          obj.genres[0]?.name
        )}
      </div>
      <p className={styles.duration}>{hours < 0 ? obj.movieLength + ' мин' : hours + ' ч ' + min + ' мин'}</p>
    </div>
  );
};
