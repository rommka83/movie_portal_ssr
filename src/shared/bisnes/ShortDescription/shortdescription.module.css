.root {
  margin-top: 10px;
}
.root > p {
  font-size: 12px;
  color: var(--gray-72);
}
.year–°ountry {
  margin-bottom: 5px;
}
.genre {
  margin-bottom: 5px;
}
.genreBookmarks {
  justify-content: center;
}
