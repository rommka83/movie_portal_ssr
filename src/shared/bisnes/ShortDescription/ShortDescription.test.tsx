import renderer from 'react-test-renderer';
import { ShortDescription } from './ShortDescription';

const film = {
  id: 0,
  name: '',
  enName: null,
  year: 0,
  description: null,
  shortDescription: null,
  rating: null,
  movieLength: 0,
  ageRating: null,
  poster: {
    url: '',
    previewUrl: '',
  },
  videos: {
    trailers: [],
    teasers: [],
  },
  genres: [{ name: 'sss' }],
  countries: [],
  persons: [],
  similarMovies: null,
  facts: null,
  alternativeName: null,
};

describe('проверка ShortDescription', () => {
  test('отрисовка ShortDescription ', () => {
    const component = renderer.create(<ShortDescription obj={film} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
