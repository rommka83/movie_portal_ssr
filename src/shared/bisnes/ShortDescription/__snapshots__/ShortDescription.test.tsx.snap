// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка ShortDescription отрисовка ShortDescription  1`] = `
<div
  className="root"
>
  <p
    className="yearСountry"
  >
    1970
    , 
  </p>
  <div
    className="genre"
  >
    sss
  </div>
  <p
    className="duration"
  >
    0 ч 0 мин
  </p>
</div>
`;
